modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/swaption/SabrSwaptionPhysicalProductPricer.java

```diff
@@ -0,0 +1,257 @@
+   * @return the present value curve sensitivity of the swap product
+   */
+  public PointSensitivityBuilder presentValueSensitivity(
+      SwaptionProduct swaption,
+      RatesProvider ratesProvider,
+      SabrVolatilitySwaptionProvider volatilityProvider) {
+
+    ExpandedSwaption expanded = swaption.expand();
+    validate(ratesProvider, expanded, volatilityProvider);
+    ZonedDateTime expiryDateTime = expanded.getExpiryDateTime();
+    double expiry = volatilityProvider.relativeTime(expiryDateTime);
+    ExpandedSwap underlying = expanded.getUnderlying();
+    ExpandedSwapLeg fixedLeg = fixedLeg(underlying);
+    if (expiry < 0d) { // Option has expired already
+      return PointSensitivityBuilder.none();
+    }
+    double tenor = volatilityProvider.tenor(fixedLeg.getStartDate(), fixedLeg.getEndDate());
+    double shift = volatilityProvider.getParameters().getShift(DoublesPair.of(expiry, tenor));
+    double forward = swapPricer.parRate(underlying, ratesProvider);
+    double pvbp = swapPricer.getLegPricer().pvbp(fixedLeg, ratesProvider);
+    double strike = swapPricer.getLegPricer().couponEquivalent(fixedLeg, ratesProvider, pvbp);
+    ValueDerivatives volatilityAdj = volatilityProvider.getParameters().getVolatilityAdjoint(expiry, tenor, strike, forward);
+    boolean isCall = (fixedLeg.getPayReceive() == PayReceive.PAY);
+    // Payer at strike is exercise when rate > strike, i.e. call on rate
+    // Backward sweep
+    PointSensitivityBuilder pvbpDr = swapPricer.getLegPricer().pvbpSensitivity(fixedLeg, ratesProvider);
+    PointSensitivityBuilder forwardDr = swapPricer.parRateSensitivity(underlying, ratesProvider);
+    double shiftedForward = forward + shift;
+    double shiftedStrike = strike + shift;
+    double price = BlackFormulaRepository.price(shiftedForward, shiftedStrike, expiry, volatilityAdj.getValue(), isCall);
+    double delta = BlackFormulaRepository.delta(shiftedForward, shiftedStrike, expiry, volatilityAdj.getValue(), isCall);
+    double vega = BlackFormulaRepository.vega(shiftedForward, shiftedStrike, expiry, volatilityAdj.getValue());
+    double sign = (expanded.getLongShort() == LongShort.LONG) ? 1d : -1d;
+    return pvbpDr.multipliedBy(price * sign * Math.signum(pvbp))
+        .combinedWith(forwardDr.multipliedBy((delta + vega * volatilityAdj.getDerivative(0)) * Math.abs(pvbp) * sign));
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value sensitivity to the SABR model parameters of the swaption product.
+   * <p>
+   * The sensitivity of the present value to the SABR model parameters, alpha, beta, rho and nu.
+   * 
+   * @param swaption  the swaption product
+   * @param ratesProvider  the rates provider
+   * @param volatilityProvider  the Black volatility provider
+   * @return the sensitivity to the SABR model parameters 
+   */
+  public SwaptionSabrSensitivity presentValueSabrParameterSensitivity(
```

Comment:
Should have a consistent naming convention with the "BlackSwaptionPhysicalProductPricer" where the name is "presentValueSensitivityBlackVolatility". The parameter "BlackVolatility" or "SabrParameter" should be before or after "Sensitivity" in all name on a consistent basis.